### PR TITLE
Paytrace

### PR DESCRIPTION
**Pull request by danieltaylor-nz**

### CODE REVIEW

File: service/PayTrace/PayTracePaymentServices.xml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                            "expiration_month":"${creditCard?.expireDate ? creditCard.expireDate.substring(0, creditCard.expireDate.indexOf('/')) : null}",
+                            "expiration_year":"${creditCard?.expireDate ? creditCard.expireDate.substring(creditCard.expireDate.indexOf('/')+1) : null}"
+                        },
+                        "csc": "${cardSecurityCode ? cardSecurityCode: '999'}",
+                        "billing_address":{
+                        "name":"${postalAddress?.toName}",
+                        "street_address":"${postalAddress?.address1}",
+                        "city":"${postalAddress?.city}",
+                        "state":"${postalAddress?.stateProvinceGeoId ? postalAddress.stateProvinceGeoId.substring(postalAddress.stateProvinceGeoId.indexOf('_')+1) : null}",
+                        "zip":"${postalAddress?.postalCode}"
```
Comment (danieltaylor-nz) [inline service/PayTrace/PayTracePaymentServices.xml]: I think the above "billing_address " properties should be indented

File: service/PayTrace/PayTraceServices.xml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            <service-call name="create#mantle.account.method.PaymentGatewayResponse" out-map="context"
+                    in-map="[paymentGatewayConfigId:paymentGatewayConfigId, paymentOperationEnumId:'PgoStoreCim',
+                            paymentMethodId:paymentMethodId, transactionDate:ec.user.nowTimestamp,
+                            responseCode:responseCode, reasonCode:reasonCode, reasonMessage:reasonMessage, resultSuccess:'Y', resultError:'N']"/>
+
+            <!-- set the gatewayCimId values for Party and PaymentMethod as needed -->
+            <if condition="paymentMethod.gatewayCimId"><then>
+                <!-- have a payment profile, called updateCustomerPaymentProfileRequest, no need to do anything -->
+            </then><else-if condition="party.gatewayCimId">
+                <!-- no payment profile but we have a customer profile, called createCustomerPaymentProfileRequest -->
```
Comment (danieltaylor-nz) [inline service/PayTrace/PayTraceServices.xml]: I think this is a comment from Authorize.net implementation, please remove.

File: entity/MantlePaytraceEntities.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+along with this software (see the LICENSE.md file). If not, see
+<http://creativecommons.org/publicdomain/zero/1.0/>.
+-->
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://moqui.org/xsd/entity-definition-2.1.xsd">
+    <entity entity-name="PaymentGatewayPayTrace" package="PayTrace">
+        <field name="paymentGatewayConfigId" type="id" is-pk="true"/>
+        <field name="transactionUrl" type="text-medium"><description>Transaction URL</description></field>
+        <field name="username" type="text-medium" encrypt="true"><description>PayTrace API login (username)</description></field>
+        <field name="password" type="text-medium" encrypt="true"><description>PayTrace API login (password)</description></field>
+        <field name="integratorID" type="text-medium" encrypt="true">
```
Comment (jonesde) [inline entity/MantlePaytraceEntities.xml]: The general naming rule for Java camel case is to only upper case the first letter in acronyms and abbreviations so 'integratorID' should be integratorId' (lower case 'd' at the end)

**Comment by jonesde** [comment]
As a general comment: for the vault/CIM integration to Mantle services SECA rules are needed like those in the service/AuthorizeDotNet.secas.xml file in the AuthorizeDotNet component (framework pattern for picking up SECA files is must be in the 'service' directory in a component and much has extension '.secas.xml'; also note that these don't reload on the fly like many other files, have to restart Moqui to get changes running).

**Pull request merged**